lisa/tests/scheduler/numa_behaviour.py

```diff
@@ -0,0 +1,132 @@
+        """
+        Factory method to create a bundle using a live target
+
+        This will execute the rt-app workload described in
+        :meth:`lisa.tests.base.RTATestBundle.get_rtapp_profile`
+        """
+        plat_info = target.plat_info
+        rtapp_profile = cls.get_rtapp_profile(plat_info)
+
+        cls.run_rtapp(target, res_dir, rtapp_profile, ftrace_coll=ftrace_coll)
+
+        return cls(res_dir, plat_info)
+
+    def _get_task_cpu_df(self, task):
+        """
+        Get a DataFrame for task mirgrations
+
+        Use the sched_switch trace event to find task mirgation from one CPU to another.
+
+        :returns: A Pandas DataFrame for the task, showing the
+                  new CPU that the task was migrated to
+        """        
+        df = self.trace.df_events('sched_switch')[['next_pid', 'next_comm', '__cpu']]
+        task_id = self.trace.get_task_id(task, update=False)
+        _df = df_filter_task_ids(df, [task_id], pid_col='next_pid', comm_col='next_comm')
+        cpu_df = _df.drop_duplicates(subset='__cpu', keep='first', inplace=False)
+                
+        return cpu_df
+
+    @requires_events('sched_switch')
+    def test_task_placement(self) -> ResultBundle:
+        """
+        Test that task remained on the same core 
+
+        """
+        test_passed = True
+        failed_tasks = []
+        
+        for task in self.rtapp_tasks:
+            cpu_df = self._get_task_cpu_df(task)
+            core_migrations = len(cpu_df.index)
+            # Ideally, task with low utilization
+            # should stay on the same core and
+            # do not cross NUMA boundaries
+            if  core_migrations > 1:
+                test_passed = False
+                failed_tasks.append("{} core_migrations {}".format(task, core_migrations))
+
+        res = ResultBundle.from_bool(test_passed)
+        res.add_metric("Failed tasks", failed_tasks)
+        
+        return res
+
+class NUMA_SmallTaskPlacement(NUMABehaviour):
+    """
+    A single task with 50% utilization
+    """
+
+    task_prefix = "ntask"
+
+    @classmethod
+    def get_rtapp_profile(cls, plat_info):
+        rtapp_profile = {}
+        rtapp_profile[cls.task_prefix] = Periodic(
+            duty_cycle_pct=50,
+            duration_s=30,
+            period_ms=cls.TASK_PERIOD_MS
+        )
+
+        return rtapp_profile
+    
+class NUMA_MultipleTasksPlacement(NUMABehaviour):
+    """
+    Multiple tasks with 50% utilization
+    """
+    task_prefix = "ntask"
+    
+    @classmethod
+    def get_rtapp_profile(cls, plat_info):
+        # Four CPU's is enough to demonstrate task migration problem
+        cpu_count = min(4, plat_info["cpus-count"])        
+
+        rtapp_profile = {}
+        for cpu in range(cpu_count):            
+            rtapp_profile["{}_{}".format(cls.task_prefix, cpu)] = Periodic(
```

Comment:
FYI the task name is currently limited to 7 chars (kernel saved `comm` size minus some suffix added by rtapp itself). If you revisit the number of CPUs involved in the test and if it becomes > 10, you will end up with names like `ntask_10` which won't fit anymore. What we did in other tests is drop the `_` to avoid such issues.